core/services/relay/evm/chain_writer.go

```diff
@@ -59,36 +81,66 @@ func (w *chainWriter) SubmitTransaction(ctx context.Context, contract, method st
 		return fmt.Errorf("method config not found: %v", method)
 	}
 
-	forwarderABI := evmtypes.MustGetABI(contractConfig.ContractABI)
-
-	calldata, err := forwarderABI.Pack(methodConfig.ChainSpecificName, args...)
+	calldata, err := w.encoder.Encode(ctx, args, wrapItemType(contract, method, true))
 	if err != nil {
-		return fmt.Errorf("pack forwarder abi: %w", err)
+		return err
 	}
 
 	var checker evmtxmgr.TransmitCheckerSpec
 	if methodConfig.Checker != "" {
 		checker.CheckerType = txmgrtypes.TransmitCheckerType(methodConfig.Checker)
 	}
 
-	var sendStrategy txmgrtypes.TxStrategy = txmgr.SendEveryStrategy{}
-	if w.config.SendStrategy != nil {
-		sendStrategy = w.config.SendStrategy
-	}
-
 	req := evmtxmgr.TxRequest{
 		FromAddress:    methodConfig.FromAddress,
 		ToAddress:      common.HexToAddress(toAddress),
 		EncodedPayload: calldata,
 		FeeLimit:       methodConfig.GasLimit,
 		Meta:           &txmgrtypes.TxMeta[common.Address, common.Hash]{WorkflowExecutionID: meta.WorkflowExecutionID},
-		Strategy:       sendStrategy,
+		Strategy:       w.sendStrategy,
 		Checker:        checker,
 	}
 
 	_, err = w.txm.CreateTransaction(ctx, req)
 	if err != nil {
-		return fmt.Errorf("failed to create tx: %w", err)
+		return fmt.Errorf("%w; failed to create tx", err)
+	}
+
+	return nil
+}
+
+func (w *chainWriter) parseContracts() error {
```

Comment:
It probably makes sense to add additional tests for this function to ensure it doesn't accept invalid ABIs